CMakePresets.json

```diff
@@ -6,9 +6,69 @@
     "patch": 0
   },
   "configurePresets": [
+    {
+      "name": "base",
+      "description": "Base options",
+      "binaryDir": "${sourceDir}/build",
+      "generator": "Ninja",
+      "cacheVariables": {
+        "CMAKE_BUILD_TYPE": "Release",
+        "CMAKE_INSTALL_PREFIX": "${sourceDir}/install",
+        "CMAKE_INTERPROCEDURAL_OPTIMIZATION": "OFF",
+        "FULL_BUILD": "ON"
+      }
+    },
+    {
+      "name": "ci-linux",
+      "inherits": "base",
+      "description": "Linux options for CI build (PR and Main)"
+    },
+    {
+      "name": "ci-macos",
+      "inherits": "base",
+      "description": "MacOS options for CI build (PR and Main)",
+      "cacheVariables": {
+        "CMAKE_OSX_DEPLOYMENT_TARGET": "10.15"
+      }
+    },
+    {
+      "name": "ci-windows",
+      "inherits": "base",
+      "description": "Windows options for CI build (PR and Main)",
+      "generator": "Visual Studio 16 2019",
+      "architecture": "x64"
+    },
+    {
+      "name": "package",
+      "inherits": "base",
+      "description": "Package build base options",
+      "cacheVariables": {
+        "CMAKE_INSTALL_PREFIX": "$env{PREFIX}",
+        "CMAKE_INTERPROCEDURAL_OPTIMIZATION": "ON"
+      }
+    },
+    {
+      "name": "package-linux",
+      "inherits": "package",
+      "description": "Linux options for package build"
+    },
+    {
+      "name": "package-macos",
+      "inherits": "package",
+      "description": "MacOS options for package build",
+      "cacheVariables": {
+        "CMAKE_OSX_DEPLOYMENT_TARGET": "10.15"
```

Comment:
Duplicate of ci-macos. We could have base macos / linux / windows presets to set these variables. Or can you only inherit from a single preset?

Reply:
Yes, no double inheritance as far as I can tell. :(